### PR TITLE
Logging bug fix

### LINKED ISSUE #502: Dynamic Log filtering doesn't appear to be working
> Question about Log filtering, I have ILogger setup and it's outputs to the dynamic console, and "Configure Logging Levels" showing, when you output from controller how do you filter on Error and critical only ?  because I changed as it show but I am still getting the info messages.
<sub>[View in Slack](https://vmware.slack.com/archives/C1VKUERFH/p1604681075122400)</sub>

### PR DESCRIPTION
**Pull request by hananiel**
Resolves #502

### CODE REVIEW

File: src/Logging/src/SerilogDynamicLoggerCore/SerilogBuilderExtensions.cs [hunk 1]
```diff
@@ -32,6 +33,12 @@ public static ILoggingBuilder AddSerilogDynamicConsole(this ILoggingBuilder buil
                 throw new InvalidOperationException("An IDynamicLoggerProvider has already been configured! Call 'AddSerilogDynamicConsole' earlier in program.cs (Before AddCloudFoundryActuators()) or remove duplicate IDynamicLoggerProvider entries.");
             }
 
+            var defaultConsoleDescriptor = builder.Services.FirstOrDefault(d => d.ImplementationType == typeof(ConsoleLoggerProvider));
+            if (defaultConsoleDescriptor != null)
+            {
+                builder.Services.Remove(defaultConsoleDescriptor);
+            }
+
```
Comment (TimHess) [inline src/Logging/src/SerilogDynamicLoggerCore/SerilogBuilderExtensions.cs:41]: As-is, this seems like a change that might be too extreme for a patch release (actively removing something that wasn't added by our code). Can we make it opt-in and switch it to be default behavior down the road?

Comment (hananiel) [reply src/Logging/src/SerilogDynamicLoggerCore/SerilogBuilderExtensions.cs:41]: We are just making it consistent with DynamicConsole -  https://github.com/SteeltoeOSS/Steeltoe/blob/4a396adf1069ccd1ad3bc8603719569579efae7b/src/Logging/src/DynamicLogger/DynamicLoggingBuilder.cs#L33
But if you feel strongly I can add a flag and allow opt in.

Comment (TimHess) [reply src/Logging/src/SerilogDynamicLoggerCore/SerilogBuilderExtensions.cs:41]: I'm in favor of it being this way -- I'm just hesitating on adding it _in a patch release_ where it is on-by-default and there's no opt-out.

Comment (hananiel) [reply src/Logging/src/SerilogDynamicLoggerCore/SerilogBuilderExtensions.cs]: Added an opt-out to preserve default logger.

File: src/Logging/src/SerilogDynamicLoggerCore/SerilogBuilderExtensions.cs [hunk 2]
```diff
@@ -20,18 +21,28 @@ namespace Steeltoe.Extensions.Logging.SerilogDynamicLogger
     public static class SerilogBuilderExtensions
     {
         /// <summary>
-        /// Add Steeltoe logger wrapped in a <see cref="IDynamicLoggerProvider"/> that supports
+        /// Add Serilog logger wrapped in a <see cref="IDynamicLoggerProvider"/> that supports
         /// dynamically controlling the minimum log level via management endpoints
         /// </summary>
         /// <param name="builder">The <see cref="ILoggingBuilder"/> for configuring the LoggerFactory  </param>
+        /// <param name="preserveDefaultConsole">When true, do not remove the default console</param>
```
Comment (TimHess) [inline src/Logging/src/SerilogDynamicLoggerCore/SerilogBuilderExtensions.cs:28]:
```suggestion
        /// <param name="preserveDefaultConsole">When true, do not remove Microsoft's ConsoleLoggerProvider</param>
```

**Review by TimHess — DISMISSED** [review]

**Review by TimHess — APPROVED** [review]

**Pull request merged**